Gems/Atom/RHI/DX12/Code/Source/RHI/TransientAttachmentPool.cpp

```diff
@@ -30,7 +30,7 @@ namespace AZ
         {
             Device& device = static_cast<Device&>(deviceBase);
 
-            const uint32_t ObjectCacheSize = 256;
+            const uint32_t ObjectCacheSize = 512;
```

Comment:
Should we data drive this value using the [SettingsRegistry](https://www.docs.o3de.org/docs/user-guide/settings/developer-documentation/#get-started-with-the-settings-registry)?

That would allow users to alter the value without needing to re-compile.
```suggestion
            // lowercased now as this is no longer a Constant per the coding standard
            uint32_t objectCacheSize = 512;
            if (auto settingsRegistry = AZ::SettingsRegistry::Get();
                settingsRegistry != nullptr);
            {
                constexpr const char* ObjectCacheSizeKey = "/O3DE/Atom/RHI/Settings/ObjectCacheSize";
                settingsRegistry->Get(objectCacheSize, ObjectCacheSizeKey)
            }
```

A user could then toss a JSON file into either their `<project-path>/user/Registry` or their `~/.o3de/Registry` directory and to have a per user override of the setting.

Reply:
yeah we can data drive this and we do this with other settings within RHI. I guess this was not converted over. As an example if you look at C:\GitHub\o3de\Gems\Atom\RHI\Registry\Platform\Windows you will see .setreg files where we allow developers to provide platform specific data via PlatformLimitsDescriptor. The only issue is that it will increase the scope of this work.

Reply:
It seems like this value is overridden in Vk/metal backends as well (not ideal I know) so they will need to be updated too. 

Vk - AliasedAttachmentAllocator* TransientAttachmentPool::CreateAllocator
Metal - RHI::ResultCode TransientAttachmentPool::InitInternal

If you search for the above functions in Atom folder you will find them.

Reply:
Data driving this is definitely a good idea. I can attempt to add this functionality, and if i have questions while doing it I will ask you in the Discord server.

Reply:
That sounds good with me.